addons/account/migrations/9.0.1.1/pre-migration.py

```diff
@@ -165,6 +166,210 @@ def map_account_tax_template_type(cr):
+        "account_move_line", "balance", "numeric",
+        "Technical field holding the debit - credit in order to open"
+        " meaningful graph views from reports"
+    )
+    create_field(
+        cr,
+        "account_move_line", "debit_cash_basis", "numeric",
+        "Debit Cash Basis"
+    )
+    create_field(
+        cr,
+        "account_move_line", "credit_cash_basis", "numeric",
+        "Credit Cash Basis"
+    )
+    create_field(
+        cr,
+        "account_move_line", "balance_cash_basis", "numeric",
+        "Balance Cash Basis"
+    )
+    # user_type_id has 'old' name user_type, but that is not in database:
+    create_field(
+        cr,
+        "account_move_line", "user_type_id", "int4",
+        "User Type Id"
+    )
+    # following field took hours to recompute.
+    create_field(
+        cr,
+        "account_invoice_line", "price_subtotal_signed", "numeric",
+        "Total amount in the currency of the company,"
+        " negative for credit notes."
+    )
+    # Set fields on account_move
+    # matched_percentage will be set to 0.0, but be filled in migration
+    #     of reconciliations.
+    openupgrade.logged_query(
+        cr,
+        """\
+        UPDATE account_move
+         SET currency_id = subquery.currency_id,
+             amount = subquery.amount,
+             matched_percentage = 0.0
+         FROM (SELECT
+                am.id as id,
+                rc.currency_id as currency_id,
+               sum(aml.debit) as amount
+            FROM account_move am
+            JOIN res_company rc ON rc.id = am.company_id
+            LEFT OUTER JOIN account_move_line aml ON aml.move_id = am.id
+            GROUP BY am.id, rc.currency_id
+        ) as subquery
+        WHERE account_move.id = subquery.id
+        """
+    )
+    # cash basis fields depend om matched_percentage, which will be filled
+    # by reconciliation migration. For now will be filled with values that
+    # are correct if associated journal is not for sale or purchase.
+    openupgrade.logged_query(
+        cr,
+        """\
+        UPDATE account_move_line
+        SET amount_residual = 0.0,
+            amount_residual_currency = 0.0,
+            reconciled = False,
+            company_currency_id = subquery.company_currency_id,
+            balance = subquery.balance,
+            debit_cash_basis = subquery.debit,
+            credit_cash_basis = subquery.credit,
+            balance_cash_basis = subquery.balance,
+            user_type_id = subquery.user_type_id
+        FROM (
+            SELECT
+                aml.id as id,
+                rc.currency_id as company_currency_id,
+                (aml.debit - aml.credit) as balance,
+                aml.debit as debit,
+                aml.credit as credit,
+                aa.user_type as user_type_id
+            FROM account_move_line aml
+            JOIN res_company rc ON rc.id = aml.company_id
+            JOIN account_account aa ON aa.id = aml.account_id
```

Comment:
I need a subquery as far as I understand how UPDATE FROM works that has one line for each of the lines to be updated. If that is not needed I am mistaken.